### PR TITLE
Add osqp-matlab

### PR DESCRIPTION
**Pull request by traversaro**
Add support for compiling the osqp-matlab bindings if ROBOTOLOGY_ENABLE_DYNAMICS and ROBOTOLOGY_USES_MATLAB are enabled. 

This PR is not ready to be merged as it depends on https://github.com/dic-iit/osqp-matlab-cmake-buildsystem/pull/1/files to be merged and the release v0.6.2.0 to be created, but it can be already reviewed. 

Fix https://github.com/robotology/robotology-superbuild/issues/575 .

### CODE REVIEW

File: cmake/Buildosqp-eigen.cmake
Comment (nunoguedelha) [inline cmake/Buildosqp-eigen.cmake:1]: I guess this is a typo: filename should be `Buildosqp-matlab.cmake`, unless I missed something.

File: cmake/Buildosqp-eigen.cmake
Comment (nunoguedelha) [inline cmake/Buildosqp-eigen.cmake:10]: This is a left-over I guess: repo should be `dic-iit/osqp-matlab-cmake-buildsystem`.

File: cmake/Buildosqp-eigen.cmake
Comment (nunoguedelha) [inline cmake/Buildosqp-eigen.cmake:11]: ither change the branch there to `master`, or here to `main`. Out of curiosity, why `main`?

File: releases/2021.02.yaml
Comment (nunoguedelha) [inline releases/2021.02.yaml:186]: `osqp-matlab` disappeared after the merge from master. Was this intended?

**Review by nunoguedelha — CHANGES_REQUESTED** [review]

File: cmake/Buildosqp-eigen.cmake
Comment (traversaro) [reply cmake/Buildosqp-eigen.cmake:1]: Yes!

File: cmake/Buildosqp-eigen.cmake
Comment (traversaro) [reply cmake/Buildosqp-eigen.cmake]: Fixed.

File: releases/2021.02.yaml
Comment (traversaro) [reply releases/2021.02.yaml:186]: Yes. The PR was opened before the 2021.02 release, but as the 2021.02 release has passed, it does not make sense to add a release for osqp-matlab here. Furthermore, we now have the latest releases file: https://github.com/robotology/robotology-superbuild/pull/596 , so once osqp-matlab has a release we can add it there.

File: cmake/Buildosqp-eigen.cmake
Comment (traversaro) [reply cmake/Buildosqp-eigen.cmake]: I changed to main. main is the new default branch name in GitHub : https://github.com/github/renaming#new-repositories-use-main-as-the-default-branch-name .

Comment (nunoguedelha) [reply cmake/Buildosqp-eigen.cmake]: Thx! good to know.

File: releases/2021.02.yaml
Comment (nunoguedelha) [reply releases/2021.02.yaml:186]: Indeed, I had seen that one. Great.

Comment (nunoguedelha) [reply releases/2021.02.yaml:186]: > once osqp-matlab has a release we can add it there.

Do we do it in a new issue or we hold this PR until then?

Comment (traversaro) [reply releases/2021.02.yaml:186]: I think we can review https://github.com/dic-iit/osqp-matlab-cmake-buildsystem/pull/2 and then I will release a first version of osqp-matlab so that we can add its release here.

Comment (nunoguedelha) [reply releases/2021.02.yaml:186]: Ok. I took a look at `robotologyUpdateLatestReleases.sh` and saw that in this line
https://github.com/robotology/robotology-superbuild/blob/ac6952e5f51d000a7e609e4d9684aae123db3752/scripts/robotologyUpdateLatestReleases.sh#L72
it assumes that the entry for the package exists, so if I understood it well, I guess we'll have to add it manually the first time.

Comment (nunoguedelha) [reply releases/2021.02.yaml:186]: looking at https://github.com/dic-iit/osqp-matlab-cmake-buildsystem/pull/2 now...

Comment (traversaro) [reply releases/2021.02.yaml:186]: >  it assumes that the entry for the package exists, so if I understood it well, I guess we'll have to add it manually the first time.

Yes!

**Comment by traversaro** [comment]
The `latest.releases.yml` file has been updated with the v0.6.2.0 release.

**Review by nunoguedelha — APPROVED** [review]

**Pull request merged**
